syntax/crisp_lexer.mll

```diff
@@ -11,6 +11,40 @@
 open Lexing
 open Crisp_parser
 
+exception Error
+
+let backtrack str lexbuf =
+  lexbuf.lex_curr_pos <- lexbuf.lex_curr_pos - (String.length str)
+
+let count_nl str =
+  if str <> (String.make (String.length str) '\n') then
+    failwith "Counting new lines in a string containing not only new lines"
+  else
+    String.length str (*NOTE assumes we only use unix-style newlines*)
+
+let next_line ?(nl_count=1) ?nl_str ?(nl_at_end=false) lexbuf =
+  (*this function was adapted from
+    https://realworldocaml.org/v1/en/html/parsing-with-ocamllex-and-menhir.html*)
+  let nl_count =
+    match nl_str with
+    | None -> nl_count
+    | Some str -> count_nl str in
```

Comment:
It seems that the `nl_str` parameter overrides the `nl_count` parameter, and when both `nl_count` and `nl_str` are provided to `next_line`, the former is ignored. It's worth noting this in a comment just before the definition of `next_line`, in case future users of the function don't realise that there is a dependency among its parameters. Also, to code defensively, you could throw an exception if you have that both `nl_count <> 1` and `nl_str <> None` (the default values), to make assumed relation more explicit.